internal/gps/source.go

```diff
@@ -507,14 +507,16 @@ func (sg *sourceGateway) require(ctx context.Context, wanted sourceState) (errSt
 					}
 				}
 			case sourceHasLatestVersionList:
-				var pvl []PairedVersion
-				err = sg.suprvsr.do(ctx, sg.src.sourceType(), ctListVersions, func(ctx context.Context) error {
-					pvl, err = sg.src.listVersions(ctx)
-					return err
-				})
+				if len(sg.cache.getAllVersions()) == 0 {
```

Comment:
so this conditional is kinda undermining the purpose of the FSM - it's effectively saying that EITHER the `sourceHasLatestVersionList` bit can be set, OR the cache can have a non-empty version set. it was this kind of multiple-sources-of-truth stuff in the old implementation that made me move to this model in the first place.

basically, if there's a case where this conditional isn't hit, it means there's a bug _somewhere else_, as that place should've also known to go and update `sg.srcState` accordingly.